contracts/implementations/nodes/Integration.sol

```diff
@@ -0,0 +1,287 @@
+    /// @dev Caller must be an admin
+    /// @dev It is assumed the 'Name' attribute is whitelisted in advance
+    /// @param owner The address of the new owner
+    /// @param names List of integration names
+    function mintIntegrationBatch(address owner, string[] calldata names)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(_hasRole(DEFAULT_ADMIN_ROLE, owner), "Owner must be an admin");
+
+        IntegrationStorage.Storage storage s = IntegrationStorage.getStorage();
+
+        uint256 newTokenId;
+        string memory name;
+        for (uint256 i = 0; i < names.length; i++) {
+            name = names[i];
+
+            require(
+                s.integrationNameToNodeId[name] == 0,
+                "Integration name already registered"
+            );
+
+            newTokenId = INFT(s.idProxyAddress).safeMint(owner);
+
+            s.integrationNameToNodeId[name] = newTokenId;
+            s.nodeIdToIntegrationName[newTokenId] = name;
+
+            emit IntegrationNodeMinted(newTokenId, owner);
+        }
+    }
+
+    /// @notice Mints an integration
+    /// @dev Caller must be an admin
+    /// @param owner The address of the new owner
+    /// @param name Name of the integration
+    /// @param attrInfoPairList List of attribute-info pairs to be added
+    function mintIntegration(
+        address owner,
+        string calldata name,
+        AttributeInfoPair[] calldata attrInfoPairList
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        IntegrationStorage.Storage storage s = IntegrationStorage.getStorage();
+        require(!s.controllers[owner].integrationMinted, "Invalid request");
+        require(
+            s.integrationNameToNodeId[name] == 0,
+            "Integration name already registered"
+        );
+
+        address idProxyAddress = s.idProxyAddress;
+
+        s.controllers[owner].isController = true;
+        s.controllers[owner].integrationMinted = true;
+
+        uint256 newTokenId = INFT(idProxyAddress).safeMint(owner);
+
+        s.integrationNameToNodeId[name] = newTokenId;
+        s.nodeIdToIntegrationName[newTokenId] = name;
+
+        _setInfos(newTokenId, attrInfoPairList);
+
+        emit IntegrationNodeMinted(newTokenId, msg.sender);
+    }
+
+    /// @notice Add infos to node
+    /// @dev attributes must be whitelisted
+    /// @param tokenId Node id where the info will be added
+    /// @param attrInfoList List of attribute-info pairs to be added
+    function setIntegrationInfo(
+        uint256 tokenId,
+        AttributeInfoPair[] calldata attrInfoList
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(
+            INFT(IntegrationStorage.getStorage().idProxyAddress).exists(
+                tokenId
+            ),
+            "Invalid integration node"
+        );
+        _setInfos(tokenId, attrInfoList);
+    }
+
+    /**
+     * @notice Verify if an address is allowed to own an integration node and set as minted
```

Comment:
Can we leave a little note about who calls this? I know the modifier kinda lets you in on it.

Reply:
Let us be exemplary documenters